nautobot/extras/graphql/types.py

```diff
@@ -1,9 +1,31 @@
+import graphene
+from graphene_django import DjangoObjectType
 import graphene_django_optimizer as gql_optimizer
 
 from nautobot.extras.models import Status, Tag, DynamicGroup
 from nautobot.extras.filters import StatusFilterSet, TagFilterSet, DynamicGroupFilterSet
 
 
+class NautobotObjectType(DjangoObjectType):
+    url = graphene.String()
+
+    def resolve_url(self, args):
+        return self.get_absolute_url(api=True)
+
+    class Meta:
+        abstract = True
+
+
+class OptimizedNautobotObjectType(gql_optimizer.OptimizedDjangoObjectType):
+    url = graphene.String()
+
+    def resolve_url(self, args):
+        return self.get_absolute_url(api=True)
+
+    class Meta:
+        abstract = True
```

Comment:
Feels like these would belong better in `nautobot.core.graphql.types` since they're part of the core graphql functionality in nautobot.